core/src/main/java/org/apache/calcite/sql/dialect/MysqlSqlDialect.java

```diff
@@ -196,6 +196,14 @@ public MysqlSqlDialect(Context context) {
               type.getSqlTypeName(),
               SqlParserPos.ZERO),
           SqlParserPos.ZERO);
+    case UINTEGER:
```

Comment:
One thing to note is that StarRocks/Doris dialect inherits from MysqlDialect, so we need to make sure they are consistent with MySQL regarding UNSIGNED types.

Reply:
> One thing to note is that StarRocks/Doris dialect inherits from MysqlDialect, so we need to make sure they are consistent with MySQL regarding UNSIGNED types.

Good point, I'll check this.

Reply:
Starrocks is consistent with MySQL, but Doris doesn't support UNSIGNED INTEGER/BIGINT, see [doc](https://doris.apache.org/docs/query-data/mysql-compatibility?_highlight=unsigned#%E6%95%B0%E5%AD%97%E7%B1%BB%E5%9E%8B).

Reply:
this looks suspicious; why would UBIGINT be converted to UNSIGNED? This suggests information loss.  How about the other two unsigned types?

Reply:
For unsigned types, based on Mysql's [cast](https://dev.mysql.com/doc/refman/8.0/en/cast-functions.html#function_cast), Mysql always produces an unsigned [BIGINT](https://dev.mysql.com/doc/refman/8.0/en/integer-types.html) value.
The semantics is different in Calcite if we wanna cast to UTINYINT/USMALLINT/UINTEGER, how about throwing exception for those types during casting like @NobiGo said.